addons/event_sms/models/event_registration.py

```diff
@@ -1,13 +1,63 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import api, models
+from odoo.addons.phone_validation.tools import phone_validation
 
 
 class Registration(models.Model):
     _inherit = 'event.registration'
 
+    @api.model_create_multi
+    def create(self, values_list):
+        """ Override to try to format mobile or phone numbers, as it is used
+        notably to send reminders. Having international numbers eases process
+        of preparing and sending SMS. """
+        registrations = super().create(values_list)
+        for registration in registrations.filtered(
+            lambda reg: reg.mobile or reg.phone
+        ):
+            for fname in {'mobile', 'phone'}:
+                new_number = registration._phone_format(fname)
+                if new_number and new_number != registration[fname]:
+                    registration[fname] = new_number
+
+        return registrations
+
     def _sms_get_number_fields(self):
         """ This method returns the fields to use to find the number to use to
         send an SMS on a record. """
         return ['mobile', 'phone']
+
+    def _phone_format(self, fname):
+        """ Tries a formatting of mobile / phone numbers, based on partner or
+        event (or event company) country. Assumption is that either event is
+        international and people enters international-formatted numbers,
+        either event is national and people enters local numbers.
+
+        :param str fname: field name (mobile or phone)
+
+        :return: formatted or original number if cannot format (see 'phone_format'
+        and 'phone_parse' for error management)."""
+        self.ensure_one()
+
+        if fname not in self._sms_get_number_fields():
+            return False
```

Comment:
Are we checking this just in case it changes in the future or is there an override that removes it?

Maybe check the field is also valid for `self.partner_id` too before grabbing the value either way?

Reply:
Just to be a bit defensive... but not really necessary strictly speaking. Added check on both model, just because we target stable.